packages/apps/app-dashboard/src/utils/user-dashboard/getAgentPkps.ts

```diff
@@ -55,29 +56,51 @@ export async function getAgentPKPs(userAddress: string): Promise<AgentAppPermiss
       (pkp) => pkp.ethAddress.toLowerCase() !== userAddress.toLowerCase(),
     );
 
-    // For each agent PKP, fetch its appIds and categorize them
-    const pkpPermissionPromises = agentPKPs.map(async (pkp) => {
-      const client = getClient({ signer: readOnlySigner });
-      const appIds = await client.getAllPermittedAppIdsForPkp({
-        pkpEthAddress: pkp.ethAddress,
-        offset: '0',
-      });
+    // Fetch all permitted apps for all agent PKPs in a single call
+    const client = getClient({ signer: readOnlySigner });
+    const pkpEthAddresses = agentPKPs.map((pkp) => pkp.ethAddress);
 
-      return { pkp, appIds };
+    const permittedAppsArray = await client.getPermittedAppsForPkps({
```

Comment:
`getPermittedAppsForPkps` [only returns](https://github.com/LIT-Protocol/Vincent/blob/fbe69a0966b24b05abfc955213087c139aa90873/packages/libs/contracts-sdk/src/internal/user/UserView.ts#L119-L123) up to the `pageSize`, where the default is `50`. So if the User has more than 50 permitted apps for any of `pkpEthAddresses` then the 51 onward would not be returned

What needs to happen and should be built into the contracts SDK, is you need to loop over each ETH address and check if the number of returned Apps is equal to the page size, then you need to make a subsequent request where the `offset` is `pageSize` so you can check if there are any more Apps - this process would be repeated until the number of returned apps for each address is < `pageSize`